aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -178,4 +164,207 @@ Tensor mm_cuda(const Tensor& self, const Tensor& mat2) {
   return addmm_out_cuda_impl(result, result, self, mat2, 0, 1);
 }
 
+Tensor& addmm_out_cuda(Tensor &out, const Tensor &self,
+                        const Tensor &mat1, const Tensor &mat2,
+                        Scalar beta, Scalar alpha) {
+  {
+    at::NoNamesGuard guard;
+    Tensor& result = addmm_out_cuda_impl(out, self, mat1, mat2, beta, alpha);
+  }
+  at::namedinference::propagate_names_for_addmm(out, mat1, mat2, self);
+  return out;
+}
+
+Tensor addmm_cuda(const Tensor& self, const Tensor& mat1, const Tensor& mat2,
+                  Scalar beta, Scalar alpha) {
+  Tensor out = at::empty({0}, self.options());
+  addmm_out_cuda(out, self, mat1, mat2, beta, alpha);
+  return out;
+}
+
+Tensor& addmm__cuda(Tensor& self, const Tensor& mat1, const Tensor& mat2,
+                    Scalar beta, Scalar alpha) {
+  addmm_out_cuda(self, self, mat1, mat2, beta, alpha);
+  return self;
+}
+
+template<typename scalar_t>
+void addr_impl_ger_cuda(Tensor &out, const Tensor &self,
+                        const Tensor& vec1, const Tensor& vec2,
+                        scalar_t alpha, scalar_t beta) {
+  static_assert(std::is_same<scalar_t, float>::value ||
```

Comment:
Is there a reason not to use `TORCH_CHECK` here?

Reply:
To make the check at compile time rather than runtime